docs/spark/spark-configuration.md

```diff
@@ -185,4 +185,5 @@ df.write
 | snapshot-property._custom-key_    | null            | Adds an entry with custom-key and corresponding value in the snapshot summary  |
 | fanout-enabled       | false        | Overrides this table's write.spark.fanout.enabled  |
 | check-ordering       | true        | Checks if input schema and table schema are same  |
-
+| isolation-level | null | Desired isolation level for Dataframe overwrite operations.  Null => no checks, `serializable` => check for concurrent inserts or deletes in destination partitions, `snapshot` => checks for concurrent deletes in destination partitions. |
+| validate-from-snapshot-id | null | If isolation level is set, identifies base snapshot from which to check concurrent conflicts. If null, check from the first un-expired snapshot.  Obtain via `Spark3Util.loadIcebergTable().currentSnapshot().snapshotId()` |
```

Comment:
I don't think that `Spark3Util` is intended for public consumption. Plus, what table does this load?

Reply:
I've definitely been recommending that folks use this method to get table references from Spark, do you think we should be making the function private? For example when using the programmatic Action API, Spark3Util.loadIcebergTable is much less error prone than creating the Catalog class (like HiveCatalog) directly and loading the table that way.

Reply:
Yea internally we have been recommending this API, due to issues in using tables loaded by other catalogs like:  https://github.com/apache/iceberg/issues/4192, https://github.com/apache/iceberg/issues/4086.

Somehow loading a catalog with CatalogUtil giving an arbitrary name not matching the one configured in Spark, causes issue in Spark table actions like RewriteDataFiles that need the right catalog name.  

Though it should not affect this case.

Reply:
I think the main thing is that we don't guarantee anything about this API.

We should really document what people can rely on with guarantees!

Reply:
That's true, maybe we should get some Annotations? Or separate functions we consider to be for end users into a separate file?

Reply:
We were just talking about this leading up to discussion at the community sync. In my experience, annotations tend to be a (1) lot of work, (2) out of date, and (3) ignored because they're almost always inaccurate.

I think we should consider the API module stable and make semver guarantees about it, and not point end users to classes in any of the runtime Jars.

That still leaves a problem: what do we do about the classes in core that are used by engines to integrate? I think we probably want to learn more about what's there and have a reasonable policy, like we will remove things there but will go through a deprecation cycle over a release or two first.

Reply:
For this issue specifically, I think we should add a way to do common things in different environments. If Spark is present, then we should have some `Actions.get()` class that uses the current Spark context and `Catalogs.get(String name)` that gets the right catalog from Spark.